include/beast/http/basic_parser_v1.hpp

```diff
@@ -412,6 +431,24 @@ class basic_parser_v1
     needs_eof(std::false_type) const;
 
     template<class C>
+    class has_on_start_t
+    {
+        template<class T, class R =
+            decltype(std::declval<T>().on_start(
+                std::declval<error_code&>()),
+                    std::true_type{})>
+        static R check(int);
```

Comment:
We know `R` must be `std::true_type` here. We could simplify this trait by making `R` an unnamed template param and removing the `std::true_type` from the template and moving it to the function return type. Commit showing what I have in mind: https://github.com/seelabs/Beast/commit/dca9d2098380ff4c10d019597b0f3eafa4d25390

No change needed (and what you have here matches what you're done in other places. More of a comment for future traits).